packages/@lwc/synthetic-shadow/src/faux-shadow/shadow-root.ts

```diff
@@ -76,7 +76,7 @@ defineProperty(Node.prototype, ShadowRootResolverKey, {
     set(this: Node, fn: ShadowRootResolver) {
         (this as any)[ShadowResolverPrivateKey] = fn;
         // TODO [#1164]: temporary propagation of the key
-        setNodeOwnerKey(this, (fn as any).nodeKey);
+        setNodeOwnerKey(this, (fn as any)?.nodeKey);
```

Comment:
```suggestion
        if (!isUndefined(fn)) {
          setNodeOwnerKey(this, (fn as any).nodeKey);
        }
```

I can't think of why we would need to call `setNodeOwnerKey` if the `fn` is `undefined`. This is just setting a property to `undefined`, which is useless.

Reply:
This also takes care of a long standing bug(W-6740579) that has cause gacks before. @abdulsattar please mark that bug as fixed once you get this in.